packages/adapters-library/readme-new-adapter.md

```diff
@@ -0,0 +1,164 @@
+- `getRewardPositions`
+- `getRewardWithdrawals`
+- `getExtraRewardPositions`
+- `getExtraRewardWithdrawals`
+
+## 4 Create snapshot tests
+The last step involves creating tests to validate the adapter results and ensure that future changes don't break it.
+
+Inside the protocol folder, there should be a folder called tests and a file called `testCases.ts` (e.g., `src/adapters/your-protocol-id/tests/testCases.ts`). This file exports an array of test cases.
+
+You need to specify the `chain` and `method` for all test cases. Optionally, you can specify a `key` to identify the file, but this is only required if you write more than one test for the same `method` and `chain`.
+
+### 4.1 Positions
+To get a snapshot of the positions, you need to set `method: 'positions'` and provide an input field with the `userAddress`. Optionally, you can specify a `blockNumber`, but this is not required, and the latest will be used and recorded if it's left empty. This will run for all the products of this protocol.
+
+Example:
+```
+{
+    chainId: Chain.Ethereum,
+    method: 'positions',
+    input: {
+      userAddress: '0x161D61e30284A33Ab1ed227beDcac6014877B3DE',
+    },
+  }
+```
+
+### 4.2 Profits
+To get a snapshot of the profits, you need to set `method: 'profits'` and provide an `input` field with the `userAddress` and, optionally, `timePeriod`, which will default to one day if left empty. Optionally, you can specify a `blockNumber`, but this is not required, and the latest will be used and recorded if it's left empty. This will run for all the products of this protocol.
+
+Example:
+```
+{
+    chainId: Chain.Ethereum,
+    method: 'profits',
+    input: {
+      userAddress: '0x161D61e30284A33Ab1ed227beDcac6014877B3DE',
+    },
+  }
+```
+
+### 4.3 Deposits/Withdrawals/Repays/Borrows
+To get a snapshot of any of these methods, set `method: 'deposits' | 'withdrawals' | 'repays' | 'borrows'`. The input field of these methods requires additional parameters to work, including `userAddress`, `fromBlock`, `toBlock`, `protocolTokenAddress`, and `productId`.
+
+Example:
+```
+{
+    chainId: Chain.Ethereum,
+    method: 'deposits',
+    input: {
+      fromBlock: 198188138,
+      toBlock: 200597430,
+      userAddress: '0xbc0a54c02a1e80c8e25e8173a8a80baf116205b5',
+      protocolTokenAddress: '0x3bAa857646e5A0B475E75a1dbD38E7f0a6742058',
+      productId: 'supply',
+    },
+  },
+```
+
+### 4.4 TVL
+To get a snapshot of the TVL implementation, set `method: 'tvl'` and specify the protocol tokens with `filterProtocolToken: ['protocol-token-address1', 'protocol-token-address2']`. Optionally, you can specify a `blockNumber`, but this is not required, and the latest will be used and recorded if it's left empty. This will run for all the products of this protocol.
+
+Example:
+```
+{
+    chainId: Chain.Ethereum,
+    method: 'tvl',
+    filterProtocolTokens: ['0x3bAa857646e5A0B475E75a1dbD38E7f0a6742058'],
+  },
+```
```

Comment:
lets also add 

inspect your position on UI and then can you also include how to run the integration tests filter by the protocolId

Reply:
Added a few more sections for the dev ui, verifying snapshot tests and submitting a PR.